src/qibolab/platforms/utils.py

```diff
@@ -0,0 +1,93 @@
+from types import SimpleNamespace
+
+
+class Channel:
+    """Representation of physical wire connection (channel).
+
+    Name is used as a unique identifier for channels. If a channel
+    with an existing name is recreated, it will refer to the existing object.
+    Channel objects are instantiated by :class:`qibolab.platforms.platform.Platform`,
+    but their attributes are modified and used by instrument designs.
+
+    Args:
+        name (str): Name of the channel as given in the platform runcard.
+
+    Attributes:
+        ports (list): List of tuples (controller (`str`), port (`int`))
+            specifying the QM (I, Q) ports that the channel is connected.
+        qubits (list): List of tuples (:class:`qibolab.platforms.utils.Qubit`, str)
+            for the qubit connected to this channel and the role of the channel.
+        Optional arguments holding local oscillators and related parameters.
+        These are relevant only for mixer-based insturment designs.
+    """
+
+    instances = {}
+
+    def __new__(cls, name):
+        if name is None:
+            return None
+
+        if name not in cls.instances:
+            new = super().__new__(cls)
+            new.name = name
+
+            new.ports = []
+            new.qubits = []
+
+            new.local_oscillator = None
+            new.lo_frequency = 0.0
+            new.lo_power = 0.0
```

Comment:
Since you are adding all these properties to every instance, and doing essentially nothing else, promote this to a trivial `dataclass`, without any method (possibly just with your own custom `__str__`).